src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/QualityScoreDistributionSpark.java

```diff
@@ -0,0 +1,181 @@
+    public boolean INCLUDE_NO_CALLS = false;
+
+    @VisibleForTesting
+    static final class Counts implements Serializable{
+        private static final long serialVersionUID = 1L;
+        private final long[] qCounts;
+        private final long[] oqCounts;
+        private final boolean includeNoCalls;
+
+        Counts(final boolean includeNoCalls){
+           qCounts = new long[128];
+           oqCounts = new long[128];
+           this.includeNoCalls = includeNoCalls;
+        }
+
+        /**
+         * Adds a read to this count object by increasing counts of the base qualities.
+         */
+        Counts addRead(final GATKRead read) {
+            final byte[] bases = read.getBases();
+            final byte[] quals = read.getBaseQualities();
+            final byte[] oq    = ReadUtils.getOriginalBaseQualities(read);
+
+            final int length = quals.length;
+
+            for (int i=0; i<length; ++i) {
+                if (includeNoCalls || !SequenceUtil.isNoCall(bases[i])) {
+                    qCounts[quals[i]]++;
+                    if (oq != null) {
+                        oqCounts[oq[i]]++;
+                    }
+                }
+            }
+            return this;
+        }
+
+        /**
+         * Adds the other count object into this one by adding the base quality counts.
+         */
+        Counts merge(final Counts counts2) {
+            for (int i = 0; i < counts2.qCounts.length; i++) {
+                this.qCounts[i] += counts2.qCounts[i];
+            }
+            for (int i = 0; i < counts2.oqCounts.length; i++) {
+                this.oqCounts[i] += counts2.oqCounts[i];
+            }
+            return this;
+        }
+
+        long[] getQualCounts(){
+            return qCounts;
+        }
+        long[] getOrigQualCounts(){
+            return oqCounts;
+        }
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        final JavaRDD<GATKRead> reads = getReads();
+        final JavaRDD<GATKRead> filteredReads = reads.filter(makeReadFilter(PF_READS_ONLY, ALIGNED_READS_ONLY));
+        final Counts result = filteredReads.aggregate(new Counts(INCLUDE_NO_CALLS),
+                (counts, read) -> counts.addRead(read),
+                (counts1, counts2) -> counts1.merge(counts2));
+
+        final MetricsFile<?, Byte> metrics = makeMetrics(result);
+        saveResults(metrics, getHeaderForReads(), getReadSourceName());
+    }
+
+    //Convert the count object into a metrics object so save in a report
+    private MetricsFile<?, Byte> makeMetrics(final Counts result) {
+        // Built the Histograms out of the long[]s
+        final Histogram<Byte> qHisto  = new Histogram<>("QUALITY", "COUNT_OF_Q");
+        final Histogram<Byte> oqHisto = new Histogram<>("QUALITY", "COUNT_OF_OQ");
+
+        for (int i=0; i< result.qCounts.length; ++i) {
+            if (result.qCounts[i]  > 0) qHisto.increment( (byte) i, (double) result.qCounts[i]);
```

Comment:
it's not. it's performance optimization to avoid doing a bunch of work for nothing (code ported from Picard)